### PR TITLE
VMware: Fix ip facts gathered by vmware.py

### PR DESCRIPTION
**Pull request by julianbarnett**
##### SUMMARY

Currently, the vmware.py module does not gather network interface IP address information correctly if the host VM has a bond/team configuration. A bond interface utilizes the same mac address for its interface as the slave interfaces it contains. In this module when iterating over the interfaces, the dictionary (net_dict) containing the mac address (key) and list of IPs (value) will have an entry for the bond interface (with the correct IP), but then when it hits the same mac address while iterating, it will find the slave interface with that mac (with no IP information) and overwrite the value in the dictionary. This patch is to skip empty IP lists, create the mac address key if necessary, and then extend the the value list, and reset the final list for that key to only unique values. This ensures that the interfaces return valid IP information if found and won't be accidentally overwritten. 

Additionally, when enumerating the interfaces, "hw_eth" + num was used to identify the interfaces. This is misrepresentative, as the interface name on the target vm may not actually be eth[num]. I've replaced this to use the adapter label instead (entry.deviceInfo.label) which I think is much more representative and not as misleading. 

n/a

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

vmware.py patch

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by jbarnett1981** [comment]
What exactly failed here?

File: mdt_dynamic_inventory.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+MDT external inventory script
+=================================
+author: J Barnett 06/23/2016 01:15
+maintainer: J Barnett (github @jbarnett1981)
+'''
+
+import argparse
```
Comment (Akasurde) [inline mdt_dynamic_inventory.py:27]: Please use different PR for this change.

Comment (julianbarnett) [reply mdt_dynamic_inventory.py:27]: Yea i'm not sure why this is showing up in this PR. I'll delete it and try again

**Comment by jbarnett1981** [comment]
will resubmit

**Pull request closed without merging**